v2.1/import-data.md

```diff
@@ -29,53 +29,114 @@ $ cockroach sql --database=[database name] < statements.sql
+### Step 2. `IMPORT` the Postgres dump file
 
-1. Remove all statements from the file besides the `CREATE TABLE` and `COPY` statements.
-2. Manually add the table's [`PRIMARY KEY`](primary-key.html#syntax) constraint to the `CREATE TABLE` statement.
-  This has to be done manually because PostgreSQL attempts to add the primary key after creating the table, but CockroachDB requires the primary key be defined upon table creation.
-3. Review any other [constraints](constraints.html) to ensure they're properly listed on the table.
-4. Remove any [unsupported elements](sql-feature-support.html).
+To import the Postgres dump file, issue an `IMPORT` statement like the one below.
 
-### Import data
+You will need to look at the dump file and translate the Postgres [`CREATE TABLE`](create-table.html) statement there into an [`IMPORT TABLE`](import.html) statement CockroachDB understands.
 
-After reformatting the file, you can import it through `psql`:
+For this data set, the Postgres dump file required one edit: the type of the `gender` column had to be changed from `employees.employees_gender` to [`STRING`](string.html) since Postgres represented the employee's gender using a [`CREATE TYPE`](https://www.postgresql.org/docs/10/static/sql-createtype.html) statement that is not yet supported by CockroachDB. If you'd rather not edit the Postgres dump file by hand, you can run the following command (which was only tested for this example):
 
 {% include copy-clipboard.html %}
 ~~~ shell
-$ psql -p [port] -h [node host] -d [database] -U [user] < [file name].sql
+$ perl -i.bak -lapE 's/gender employees\.employees_gender/gender STRING/' employees.sql
+~~~
+
+This example uses S3. For a complete list of the types of cloud storage [`IMPORT`](import.html) can pull from, see [Import File URLs](import.html#import-file-urls).
+
+{% include copy-clipboard.html %}
+~~~ sql
+> IMPORT TABLE employees (
+    emp_no INT PRIMARY KEY,
+    birth_date DATE NOT NULL,
+    first_name STRING NOT NULL,
+    last_name STRING NOT NULL,
+    gender STRING NOT NULL,
+    hire_date DATE NOT NULL
+  )
+  PGDUMP DATA ('s3://your-external-storage/employees.sql?AWS_ACCESS_KEY_ID=ACCESSKEY&AWS_SECRET_ACCESS_KEY=SECRET');
+~~~
+
+Success will look like:
+
+~~~
++--------------------+-----------+--------------------+------+---------------+----------------+-------+
+|       job_id       |  status   | fraction_completed | rows | index_entries | system_records | bytes |
++--------------------+-----------+--------------------+------+---------------+----------------+-------+
+| 352938237301293057 | succeeded |                  1 |    0 |             0 |              0 |     0 |
++--------------------+-----------+--------------------+------+---------------+----------------+-------+
+(1 row)
+~~~
+
+Finally, we can verify that despite needing to edit the `gender` column's data type in the Postgres dump file, the column was populated correctly during the import:
```

Comment:
This sentence makes me think that we were expecting something to go wrong and are glad it didn't :) 
Also, I think we would want to verify the whole table was imported correctly. Would rewording the sentence to reflect that be better?

Reply:
> This sentence makes me think that we were expecting something to go wrong and are glad it didn't :)

I was expecting an issue after having to edit the Postgres dump file, and was pleasantly surprised when CockroachDB magically imported it correctly. :-)  I agree it's a bit strange.  I imagine most people would want to verify that a data type they hand-edited (which implies that it might not work) was imported correctly.  This is an area I suspect will change as `IMPORT` continues to add support for Postgres features.

> Also, I think we would want to verify the whole table was imported correctly. Would rewording the sentence to reflect that be better?

I'm not sure how to document this in the general case.  I think this will probably be specific to the user's application.  I imagine they would need to run their application and various tests under load against CRDB to really confirm that it works as expected for them.